homeassistant/components/osoenergy/services.yaml

```diff
@@ -0,0 +1,319 @@
+turn_on:
+  name: Turn on Heating
+  description: Turn on Heating for one hour or until max temperature is reached
+  target:
+    entity:
+      domain: water_heater
+  fields:
+    full_utilization:
+      name: Full Utilization
+      description: Choose if heating should be on until max temperature (True) is reached or for one hour (False)
+      required: true
+      default: false
+      example: false
+      selector:
+        boolean:
+turn_off:
+  name: Turn off Heating
+  description: Turn off Heating for one hour or until min temperature is reached
+  target:
+    entity:
+      domain: water_heater
+  fields:
+    full_utilization:
+      name: Full Utilization
+      description: Choose if heating should be off until min temperature (True) is reached or for one hour (False)
+      required: true
+      default: false
+      example: false
+      selector:
+        boolean:
+set_v40_min:
+  name: Set V40 Min
+  description: Set the minimum quantity of water at 40°C for a heater
+  target:
+    entity:
+      domain: water_heater
+  fields:
+    v40_min:
+      name: V40 Min
+      description: Minimum quantity of water at 40°C (200-350 for SAGA S200, 300-550 for SAGA S300)
+      required: true
+      example: 240
+      selector:
+        number:
+          min: 200
+          max: 550
+          step: 1
+          unit_of_measurement: L
+set_profile:
+  name: Set Heater Profile
+  description: Set temperature profile of water heater
+  target:
+    entity:
+      domain: water_heater
+  fields:
+    hour_00:
+      name: "00:00"
+      description: 00:00 Hour
+      required: true
```

Comment:
I would suggest marking all hour-fields as optional and only updating the value of the provided ones.

A use case for it would be if the user wants only to update the value of `hour_18` but not the rest. Currently, he needs to get the current data and pass it in.